examples/hello_world-rs/ta/Makefile

```diff
@@ -20,7 +20,7 @@ UUID ?= $(shell cat "../uuid.txt")
 TARGET ?= aarch64-unknown-linux-gnu
 CROSS_COMPILE ?= aarch64-linux-gnu-
 OBJCOPY := $(CROSS_COMPILE)objcopy
-LINKER_CFG := target.$(TARGET).linker=\"$(CROSS_COMPILE)ld.bfd\"
+LINKER_CFG := target.$(TARGET).linker=\"$(CROSS_COMPILE)gcc\"
```

Comment:
Using the `ld.bfd` works as well, right?

Reply:
> Using the `ld.bfd` works as well, right?

Yes, it did on cross_compiling, but not working if you compile on arm host machine, as ld.bfd defaultly trying to link to libgcc_s.so instead of libgcc.a if not cross compile(and the argument is `-lgcc_s`).
![image](https://github.com/user-attachments/assets/b4a0e6fa-64c8-4b33-bb91-5ea2f2a248b6)
and with `-Bstatic` provided, it tries to find `libgcc_s.a`, but this is not valid as it should be in the name of `libgcc.a`

Reply:
As you mentioned above, it's really the good point for solving another building issue https://github.com/apache/incubator-teaclave-trustzone-sdk/issues/135. 

BTW Could you provide some short guidance on how to build examples on ARM host machine (and the choice of linker) at https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/main/README.md#setup-building-environment
Such as adding two sections: "building on ARM" , "cross compile on x86_64"? I think it's valuable for our developers.

Reply:
> > Using the `ld.bfd` works as well, right?
> 
> Yes, it did on cross_compiling, but not working if you compile on arm host machine, as ld.bfd defaultly trying to link to libgcc_s.so instead of libgcc.a if not cross compile(and the argument is `-lgcc_s`). ![image](https://private-user-images.githubusercontent.com/11420684/398629399-b4a0e6fa-64c8-4b33-bb91-5ea2f2a248b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUxOTgwMzksIm5iZiI6MTczNTE5NzczOSwicGF0aCI6Ii8xMTQyMDY4NC8zOTg2MjkzOTktYjRhMGU2ZmEtNjRjOC00YjMzLWJiOTEtNWVhMmYyYTI0OGI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI2VDA3MjIxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjNTVhNWZlMjU1OGM3Mzk3MTZlYTdlYWE0MjNlNGVhYzE1NGY3MjAxNTVhMjdkM2JjMDdkZTQyNjkzNTQ2YWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5paANYW1Tl-BYJXDmKPBXjojhMzFr8dcWaPJsJtAlw4) and with `-Bstatic` provided, it tries to find `libgcc_s.a`, but this is not valid as it should be in the name of `libgcc.a`

@DemesneGH sorry for my previous misleading response, actually when using `ld.bfd` as linker, its argument is `-Bdynamic -lgcc_s`, the problem of `cannot find -lgcc_s` is that ld.bfd defaultly do not know how to find the libgcc_s.so in host machine (actually gcc has a bunch of undocument configs in it).

Here is how it happens:
1. we have a build.rs, so cargo will need to build a `build script binary` first, the target is the same as host machine, in x64 is **x86_64-unknown-linux-gnu**, in aarch64 is **aarch64-unknown-linux-gnu**.
2. we set linker config by  `target.aarch64-unknown-linux-gnu="aarch64-linux-gnu-ld.bfd"`, please keep in mind this change the linker of target **aarch64-unknown-linux-gnu**.
3. then run `make` to build ta, it tries to build a `build script binary` first and then use it to build our source codes, and things going bad starts from here. When building the `build script binary` it requires libunwind for backtrace, adds the requirement of `gcc_s` (yes, this is not added by our source codes but the `build.rs`), but we change the linker (and yes, the target of build script is **aarch64-unknown-linux-gnu** when your host is aarch64 host), so everything fuckup, you will get a link error as you cannot find libgcc_s.so(only gcc knows how to find it, for example, on my coolpi4B with ubuntu23, it is /lib/aarch64-linux-gnu/libgcc_s.so.1).

So my answer is, never use ld.bfd as default linker in if your target is the same as your host machine(aka not cross compiling)

> ### Why it works on x64 host?
> as what I described above, the target of build script is **x86_64-unknown-linux-gnu**, and the target of our source codes is **aarch64-unknown-linux-gnu**, we just change the linker of **aarch64-unknown-linux-gnu**, the build script is fine here.

Reply:
> As you mentioned above, it's really the good point for solving another building issue #135.
> 
> BTW Could you provide some short guidance on how to build examples on ARM host machine (and the choice of linker) at https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/main/README.md#setup-building-environment Such as adding two sections: "building on ARM" , "cross compile on x86_64"? I think it's valuable for our developers.

Actually there is no difference between the two environments when developing (same commands, same linux packages, same environments), I think I should just add a notification about the linker problem so that developers will know more about the building process (especially for the C developers, they may not know that there are `build script binary` and `source code binary`?

Reply:
How about keeping this modification and add a simple comment upon this:
```
-LINKER_CFG := target.$(TARGET).linker=\"$(CROSS_COMPILE)ld.bfd\"
+ # Configure the linker to use GCC, which works on both cross-compilation and ARM machines
+LINKER_CFG := target.$(TARGET).linker=\"$(CROSS_COMPILE)gcc\"
```
Since it will be added into each example's Makefile, I'm not sure if we need to explain the details such as `build script binary` and `source code binary` here. 
Would it be more reasonable to create a separate short doc for the building process in `docs/`?

Reply:
I think it's more preferable for `optee_utee_build` crate to consume `CROSS_COMPILE` environment variable and setup linker configuration accordingly.

And regarding documentation update, I don't think we want developers to learn about how `optee_utee_build` crate figures out linker given the `CROSS_COMPILE` path. Instead the documentation should rather be just about letting the user know how to build Rust TAs in either cross-compiled environment (x86 host) or native environment (arm64 host). I think that will just boil down to the value of `CROSS_COMPILE` variable being empty ("") for native compilation and ("path-to-gcc-cross-compiler") for cross compilation.

Reply:
I agree with @b49020, let's just make things simpler, we just tell TA developers setting their `CROSS_COMPILE` env, and that's enough.

But currently cargo do not allow change linker from with build script, so it still need to be done by Makefile.

Reply:
After further investigation, I found that `aarch64-linux-gnu-gcc` and `arm-linux-gnueabihf-gcc` exists on aarch64 hosts too, so I will just change `setup.sh` to install the right toolchains on aarch64 host, and keep all the steps all the same on different hosts (it's more platform-independent because, when using Rust, there shouldn't be any difference regardless of the host system developers on)

Reply:
Okay, that's fine. Let's do following additionally to further clarify things:

1. Add comment as @DemesneGH described above:

> How about keeping this modification and add a simple comment upon this:
> 
> ```
> -LINKER_CFG := target.$(TARGET).linker=\"$(CROSS_COMPILE)ld.bfd\"
> + # Configure the linker to use GCC, which works on both cross-compilation and ARM machines
> +LINKER_CFG := target.$(TARGET).linker=\"$(CROSS_COMPILE)gcc\"
> ```

2. Update https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/main/README.md#platforms with a subsection to say the supported host machines as `x86-64` and `arm64`.

Reply:
@b49020 How about add "**Currently, we support building on both aarch64 and amd64 host machines, and they share the same steps.**” under [Setup Building Environment](https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/main/README.md#setup-building-environment), I feel like the [Platforms section]( https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/main/README.md#platforms) more like **running and debugging** rather than **building**.

Reply:
> "Currently, we support building on both aarch64 and amd64 host machines, and they share the same steps.” under [Setup Building Environment](https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/main/README.md#setup-building-environment),

Let's rather be specific regarding host machines as output of command `uname -m` which gives you `x86_64` and `aarch64`. I don't have strong preference for the section to put this under.